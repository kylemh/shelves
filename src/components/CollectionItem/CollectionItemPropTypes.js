import PropTypes from 'prop-types';

const CollectionItemPropTypes = {
  id: PropTypes.number.isRequired,
  primaryArtistName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
};

export default CollectionItemPropTypes;
