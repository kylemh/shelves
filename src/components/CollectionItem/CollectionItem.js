import React from 'react';
import CollectionItemPropTypes from './CollectionItemPropTypes';
import styles from './CollectionItem.module.scss';

export default class CollectionItem extends React.Component {
  static propTypes = {
    ...CollectionItemPropTypes,
  };

  static defaultProps = {
    year: undefined,
  };

  render() {
    const { props } = this;

    return (
      <article className={styles.CollectionItem}>
        <h4 className={styles.title}>{props.title}</h4>
        <h5 className={styles.artist}>by {props.primaryArtistName}</h5>
        {Boolean(props.year) && <h5 className={styles.year}>{props.year}</h5>}
      </article>
    );
  }
}
