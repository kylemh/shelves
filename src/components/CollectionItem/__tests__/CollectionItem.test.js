import React from 'react';
import { shallow } from 'enzyme';
import CollectionItem from '../CollectionItem';

describe('CollectionItem', () => {
  const requiredProps = {
    id: 2595467,
    title: 'Cosmogramma',
    primaryArtistName: 'Flying Lotus',
    year: 2010,
  };

  it('should render', () => {
    const wrapper = shallow(<CollectionItem {...requiredProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
