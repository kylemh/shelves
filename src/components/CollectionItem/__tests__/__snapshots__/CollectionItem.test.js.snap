// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem should render 1`] = `
<article
  className="CollectionItem"
>
  <h4
    className="title"
  >
    Cosmogramma
  </h4>
  <h5
    className="artist"
  >
    by 
    Flying Lotus
  </h5>
  <h5
    className="year"
  >
    2010
  </h5>
</article>
`;
